diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -299,6 +313,19 @@ int DifferentialFunctionComparator::cmpOperations(
         }
     }
 
+    // If PHI nodes are compared and they have the same number of incoming
+    // values, treat them as equal for now. They will be compared at the end,
+    // after all their incoming values have been compared and matched.
+    if (isa<PHINode>(L) && isa<PHINode>(R)) {
+        auto PhiL = dyn_cast<PHINode>(L);
+        auto PhiR = dyn_cast<PHINode>(R);
+        if (PhiL->getNumIncomingValues() == PhiR->getNumIncomingValues()) {
+            needToCmpOperands = false;
+            phisToCompare.emplace_back(PhiL, PhiR);
+        }
+        return 0;
```

Comment:
This returns `0` without putting the PHIs to `phisToCompare` when the number of incoming values is different. which probably isn't the correct behavior, moreover this causes an assert in `cmpBasicBlocks` on the equality of the number of operands after `cmpOperations`. I think the `return` should be inside the `if` condition.